### ISSUE TITLE
Improve parsing of jupyter outputs

### ISSUE THREAD

**Issue by akhmerov** [executablebooks/mystmd#549]
This could even be considered a bug: when a cell defines multiple outputs, e.g. 
```python
import matplotlib.pyplot as plt
plt.plot([1,2,3,4])
```
then only the first output is shown in latex export. Suppressing the text output results in the image being shown.

![image](https://github.com/executablebooks/mystmd/assets/2069677/24f54dfe-611e-458f-ad52-dc9d17e97086)

I believe this isn't a standard way to treat jupyter outputs, and I propose to follow the logic implemented in myst-nb instead.

**Comment by stevejpurves** [comment]
```
plt.plot([1,2,3,4]);
```
would prevent that, the stream output is added from anything returned from the last expression in a cell.

**Comment by rowanc1** [comment]
I think it is possible that our logic changed around showing Jupyter outputs (in LaTeX) in the last few releases as well. @fwkoch would know best. :)

**Comment by akhmerov** [comment]
@stevejpurves I am aware that the text output is spurious, and I know how to suppress it. This issue is about the plot not actually being shown.

**Comment by fwkoch** [comment]
Hey - I think I see where this is coming from: We simplify `output` nodes to `image`/`text` when exporting to a static document (latex, pdf, docx) - that way, these libraries do not need to handle arbitrary rich notebook outputs (for better or worse...). However, the function that does this simplification only picks the first output and ignores the rest: https://github.com/executablebooks/mystmd/blob/main/packages/myst-cli/src/transforms/outputs.ts#L66

It should be pretty trivial to refactor that function to handle multiple outputs - I'll give it a go.

**Comment by akhmerov** [comment]
#117 is also related. As a comment:
- A cell can have multiple outputs, and displaying all is best
- Each output can have different mime types, in which case selecting according to the priority is appropriate (and priority usually depends on the render target (html/latex/jats/...). I think myst-nb demonstrates the logic rather well.

**Comment by fwkoch** [comment]
This issue was fixed by #558 (long time ago!) - thanks again for the bug report @akhmerov.
